addons/website/views/snippets/snippets.xml

```diff
@@ -588,12 +588,17 @@
         <t t-call="website.snippet_options_shadow_widgets"/>
     </div>
 
-    <!-- Mobile display options -->
-    <div data-js="MobileVisibility" data-selector="section, section .row > div"
+    <!-- Mobile/Desktop display options -->
+    <div data-js="DeviceVisibility" data-selector="section, section .row > div"
          data-exclude=".s_col_no_resize.row > div, .s_masonry_block .s_col_no_resize">
-        <we-button class="o_we_link o_we_mobile" data-show-on-mobile="true" data-no-preview="true"
-                   data-img="/website/static/src/img/snippets_options/mobile_invisible.svg"
-                   data-active-img="/website/static/src/img/snippets_options/mobile_visible.svg"/>
+        <we-button class="o_we_link o_we_device fa fa-fw fa-eye" data-toggle-device-visibility="current" data-no-preview="true"
+                   data-name="device_visibility_opt" data-device="both"/>
+        <we-button class="o_we_link o_we_device" data-toggle-device-visibility="both" data-no-preview="true"
+                   data-name="device_visibility_opt" data-device="no_mobile" title="Make visible on all devices"
+                   data-img="/website/static/src/img/snippets_options/mobile_invisible.svg"/>
+        <we-button class="o_we_link o_we_device" data-toggle-device-visibility="both" data-no-preview="true"
+                   data-name="device_visibility_opt" data-device="no_desktop" title="Make visible on all devices"
+                   data-img="/website/static/src/img/snippets_options/desktop_invisible.svg"/>
```

Comment:
In fact, I am not sure I like the UX of this whole feature:
1. The "eye" icon is really not straightforward.
2. When I am on desktop edition, I feel like it would be easier to say "I don't want this to show up on mobile" from there, rather than having to switch to mobile edition then click that button. Did we consider having a very explicit select with the three option: both, desktop only, mobile only ? "Two clicks instead of one" does not seem to be an argument here: it would be clearer IMO and it is not a feature you want to use for many blocks on your page anyway.
3. I don't understand how it is relevant for those items to show up in the "Invisible elements" section the way they do: if I click on them, the related element becomes visible... but would still be hidden in the current device. So I am not sure it is really relevant to allow to edit it then save and see it disappear. For me there are two use cases for this "invisible elements" section:
    - The standard case: there is an invisible element on the page which can show up depending on some visitor action. We need a button to show/hide it during edit mode to be able to edit it. Example: popups, cookie bar, the future column hover, ...
    - The "I want an element to be visible again" case: there is an element that was made invisible in the page but that the user wants now visible again. We had two cases (I think): when making the header or footer invisible via the related visibility option. I think this PR here should be a new case of this: when clicking on the "invisible element" entry of something hidden on the current device: make it visible again (by calling the snippet option).

@pierre-pde What do you think?

Reply:
For the record, logging the outcome of the discussion and the replacement requirements:
- "Device" option as last option (right after "Visibility" for elements that have that option)
- 2 buttons: mobile & desktop, each device can be hidden separately - mutually exclusive but both togglable "off".
- Click on invisible element -> toggle the visibility option (=> not hidden on any device anymore)